### PR TITLE
Generate rebel-remote.xml

### PR DESCRIPTION
**Pull request by Oja95**

### CODE REVIEW

File: src/main/java/org/zeroturnaround/javarebel/maven/GenerateRebelMojo.java [hunk 1]
```diff
@@ -308,6 +324,50 @@ else if (POM_PACKAGING.equals(packaging)) {
...
+        generateRebelRemoteXml(w);
+        getLog().info(w.toString());
+      } catch (IOException e) {
+        getLog().debug("Detected exception during 'showGenerated' : ", e);
+      }
+    }
+
+    try {
+      rebelXmlDirectory.mkdirs();
+      w = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(rebelRemoteXmlFile), CHARTSET_UTF8));
```
Comment (siim-k) [inline src/main/java/org/zeroturnaround/javarebel/maven/GenerateRebelMojo.java]: Is there any particular reason why the previous invocation result of generateRebelRemoteXml can not be used when writing the actual file?

File: src/main/java/org/zeroturnaround/javarebel/maven/util/SystemUtils.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package org.zeroturnaround.javarebel.maven.util;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+/**
+ * Util class to ensure safe string representation for using in paths and URLs.
+ */
+public final class SystemUtils {
```
Comment (siim-k) [inline src/main/java/org/zeroturnaround/javarebel/maven/util/SystemUtils.java]: As this particular piece of legacy code is duplicated now should we create an engineering task to make it into a library that could be publish publicly?

Comment (Oja95) [reply src/main/java/org/zeroturnaround/javarebel/maven/util/SystemUtils.java]: Will be addressed in further engineering task JR-10262.

File: src/main/java/org/zeroturnaround/javarebel/maven/GenerateRebelMojo.java (hunk 1 above)
Comment (Oja95) [reply src/main/java/org/zeroturnaround/javarebel/maven/GenerateRebelMojo.java]: `generateRebelRemoteXml` is currently written to take in a `Writer` in which the `rebel-remote.xml` contents are written into. Those two `generateRebelRemoteXml` invocations take in different types of `Writer` instances depending where its written.

**Pull request merged**